tests/clubs/test_views.py

```diff
@@ -0,0 +1,380 @@
+import json
+import datetime
+
+from django.test import TestCase, Client
+
+from clubs.models import Club, Membership, Tag, Event
+from users.models import Person
+
+
+class ClubTestCase(TestCase):
+    def setUp(self):
+        self.client = Client()
+
+        self.user1 = Person.objects.create_user('bfranklin', 'bfranklin@seas.upenn.edu', 'test')
+        self.user1.first_name = 'Benjamin'
+        self.user1.last_name = 'Franklin'
+        self.user1.save()
+
+        self.user2 = Person.objects.create_user('tjefferson', 'tjefferson@seas.upenn.edu', 'test')
+        self.user2.first_name = 'Thomas'
+        self.user2.last_name = 'Jefferson'
+        self.user2.save()
+
+        self.user3 = Person.objects.create_user('gwashington', 'gwashington@wharton.upenn.edu', 'test')
+        self.user3.first_name = 'George'
+        self.user3.last_name = 'Washington'
+        self.user3.save()
+
+        self.user4 = Person.objects.create_user('barnold', 'barnold@wharton.upenn.edu', 'test')
+        self.user4.first_name = 'Benedict'
+        self.user4.last_name = 'Arnold'
+        self.user4.save()
+
+        self.user5 = Person.objects.create_user('jadams', 'jadams@sas.upenn.edu', 'test')
+        self.user5.first_name = 'John'
+        self.user5.last_name = 'Adams'
+        self.user5.is_staff = True
+        self.user5.is_superuser = True
+        self.user5.save()
+
+
+    def test_superuser_views(self):
+        self.client.login(username=self.user1.username, password='test')
+
+        # create club
+        resp = self.client.post('/clubs/', {
```

Comment:
For testing, we should use Django's `reverse` method here instead of hardcoding the URL.

Reply:
I was thinking about doing that, but would we want to use absolute paths to ensure that we don't accidentally break backwards compatibility? The routers stuff is a bit tricky to work with.

The downside is that updating our tests if we ever explicitly want to change a path would be a pain. I'd assume that most test cases are written with `reverse` to test functionality instead of paths, but let me know what you think!

I think I'm leaning towards your viewpoint on this one, it might be better to refactor everything to use reverse.

Reply:
Yeah, I think the focus is on the functionality not the paths. I think using reverse is the way to go.

Reply:
Changed!